pkg/dataobj/sections/streams/builder.go

```diff
@@ -354,13 +356,33 @@ func numberColumnBuilder(pageSize int) (*dataset.ColumnBuilder, error) {
 	})
 }
 
-func encodeColumn(enc *encoder, columnType streamsmd.ColumnType, builder *dataset.ColumnBuilder) error {
+func encodeColumn(enc *columnar.Encoder, columnType ColumnType, builder *dataset.ColumnBuilder) error {
 	column, err := builder.Flush()
 	if err != nil {
 		return fmt.Errorf("flushing %s column: %w", columnType, err)
 	}
 
-	columnEnc, err := enc.OpenColumn(columnType, &column.Info)
+	// TODO(rfratto): remove explicit conversion once [dataset.Dataset] is
```

Comment:
FYI: These TODO comments about explicit conversion are resolved in later commits within this same PR